modules/sqlite/integration/wp-includes/sqlite/db.php

```diff
@@ -0,0 +1,281 @@
+			$pdo->rollBack();
+			$message  = sprintf(
+				'Error occured while creating tables or indexes...<br />Query was: %s<br />',
+				var_export( $rewritten_query, true )
+			);
+			$message .= sprintf( 'Error message is: %s', $err_data[2] );
+			wp_die( $message, 'Database Error!' );
+		}
+	}
+
+	$query_parser = null;
+	$pdo          = null;
+
+	return true;
+}
+
+/**
+ * Installs the site.
+ *
+ * Runs the required functions to set up and populate the database,
+ * including primary admin user and initial options.
+ *
+ * @param string $blog_title Site title.
+ * @param string $user_name User's username.
+ * @param string $user_email User's email.
+ * @param bool   $public Whether site is public.
+ * @param string $deprecated Optional. Not used.
+ * @param string $user_password Optional. User's chosen password. Default empty (random password).
+ * @param string $language Optional. Language chosen. Default empty.
+ *
+ * @return array Array keys 'url', 'user_id', 'password', and 'password_message'.
+ */
+function wp_install( $blog_title, $user_name, $user_email, $public, $deprecated = '', $user_password = '', $language = '' ) {
+	if ( ! empty( $deprecated ) ) {
+		_deprecated_argument( __FUNCTION__, '2.6.0' );
+	}
+
+	wp_check_mysql_version();
+	wp_cache_flush();
+	make_db_sqlite();
+	populate_options();
+	populate_roles();
+
+	update_option( 'blogname', $blog_title );
+	update_option( 'admin_email', $user_email );
+	update_option( 'blog_public', $public );
+
+	// Freshness of site - in the future, this could get more specific about actions taken, perhaps.
+	update_option( 'fresh_site', 1 );
+
+	if ( $language ) {
+		update_option( 'WPLANG', $language );
+	}
+
+	$guessurl = wp_guess_url();
+
+	update_option( 'siteurl', $guessurl );
+
+	// If not a public blog, don't ping.
+	if ( ! $public ) {
+		update_option( 'default_pingback_flag', 0 );
+	}
+
+	/*
+		* Create default user. If the user already exists, the user tables are
+		* being shared among sites. Just set the role in that case.
+		*/
+	$user_id        = username_exists( $user_name );
+	$user_password  = trim( $user_password );
+	$email_password = false;
+	if ( ! $user_id && empty( $user_password ) ) {
+		$user_password = wp_generate_password( 12, false );
+		$message       = __( '<strong><em>Note that password</em></strong> carefully! It is a <em>random</em> password that was generated just for you.', 'performance-lab' );
+		$user_id       = wp_create_user( $user_name, $user_password, $user_email );
+		update_user_option( $user_id, 'default_password_nag', true, true );
+		$email_password = true;
+	} elseif ( ! $user_id ) {
+		// Password has been provided.
+		$message = '<em>' . __( 'Your chosen password.', 'performance-lab' ) . '</em>';
+		$user_id = wp_create_user( $user_name, $user_password, $user_email );
+	} else {
+		$message = __( 'User already exists. Password inherited.', 'performance-lab' );
+	}
+
+	$user = new WP_User( $user_id );
+	$user->set_role( 'administrator' );
```

Comment:
There's some additional user-related functionality in WP core's current `wp_install()` function, which I would assume we should have here as well?

Overall, why do we override this core function in here? I may be missing something, but it seems like it's doing exactly the same (except for the above, which I believe may be an oversight rather than intention though).

Reply:
> There's some additional user-related functionality in WP core's current wp_install() function, which I would assume we should have here as well?

Thanks! I updated the function to bring it up to speed with all the changes from Core.

> Overall, why do we override this core function in here? I may be missing something, but it seems like it's doing exactly the same (except for the above, which I believe may be an oversight rather than intention though).

Yeah, that was not the intention (fixed now). This function was copied verbatim from https://github.com/aaemnnosttv/wp-sqlite-db/blob/504527912206a9403a38c86dd36af53bae10b116/src/db.php#L4821-L4903, but when fixing coding-standards issues, I removed the comment that clarified the purpose by mistake.
We override this function because we need to replace the call to Core's `make_db_current_silent` function, with a call to `perflab_sqlite_make_db_sqlite`.
I added a comment to make that clear 👍 
```php
/* SQLite changes: Replace the call to make_db_current_silent() with perflab_sqlite_make_db_sqlite(). */
```